bucketswatcher_http.go

```diff
@@ -137,31 +139,35 @@ func (bw *BucketsWatcherHttp) manualUpdateThread() {
 
 		bw.handleBuckets(ctx, buckets)
 
-		bw.lock.Lock()
+		bw.needsBucketLock.Lock()
```

Comment:
Why can't we use `statelock` here?

Reply:
Just a minor optimization so that updating state doesn't block access to the needsBucketChan as they're not related.